lib/accessories/outlet.js

```diff
@@ -62,104 +62,97 @@ function FritzOutletAccessory(platform, ain) {
             .setProps({minValue: -50})
             .on('get', this.getCurrentTemperature.bind(this))
         ;
+
+        this.services.TemperatureSensor.fritzCurrentTemperature = 20;
     }
 
+    this.services.Outlet.fritzState = false;
+    this.services.Outlet.fritzInUse = false;
+    this.services.Outlet.fritzPowerUsage = 0;
+    this.services.Outlet.fritzEnergyConsumption = 0;
+
+    this.update(); // execute immediately to get first initial values as fast as possible
     setInterval(this.update.bind(this), this.platform.interval);
 }
 
 FritzOutletAccessory.prototype.getOn = function(callback) {
     this.platform.log(`Getting ${this.type} ${this.ain} state`);
 
-    var service = this.services.Outlet;
-    callback(null, service.fritzState);
+    callback(null, this.services.Outlet.fritzState);
 
-    this.platform.fritz('getSwitchState', this.ain).then(function(state) {
-        service.fritzState = state;
-        service.getCharacteristic(Characteristic.On).setValue(state, undefined, FritzPlatform.Context);
-    });
+    this.queryOn();
 };
 
-FritzOutletAccessory.prototype.setOn = function(state, callback, context) {
-    var service = this.services.Outlet;
-    service.fritzState = state;
-
-    callback(null, state);
+FritzOutletAccessory.prototype.setOn = function(state, callback) {
+    this.platform.log(`Switching ${this.type} ${this.ain} to ` + state);
 
-    if (context == FritzPlatform.Context) {
-        return;
-    }
+    this.services.Outlet.fritzState = state;
+    this.platform.fritz(state ? 'setSwitchOn' : 'setSwitchOff', this.ain);
 
-    this.platform.log(`Switching ${this.type} ${this.ain} to ` + state);
+    callback();
```

Comment:
For sake of consistency- shouldn't callback now *always* be triggered before the actual FB call?

Reply:
Naja, der .fritz(...) call läuft ja asynchron.